addons/godot-next/references/singletons.gd

```diff
@@ -0,0 +1,15 @@
+# author: xdgamestudios
+# license: MIT
+# description: An API for accessing singletons
+# deps:
+# - singleton_cache.tres
+extends Reference
+class_name Singletons
+
+const singleton_cache = preload("res://addons/godot-next/resources/singletons/singleton_cache.tres")
+
```

Comment:
Just in case, I would also provide people with a `erase(p_value: Script)` method, that way people can optionally destroy and re-instance their singletons if they feel they need to. Would rather give people the option than force them to be stuck with it.

Reply:
Dealt with it already and added some functionality....